pkg/client/client.go

```diff
@@ -0,0 +1,35 @@
+package client
+
+// Provides a common client for functions outside of controllers to use.
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var commonClient client.Client
+
+func SetClient(c client.Client) {
+	commonClient = c
+}
+
+func NewClientFromConfig(cfg *rest.Config) (c client.Client, err error) {
+	commonClient, err = client.New(cfg, client.Options{})
```

Comment:
here is just Go question: I had trouble using `:=` overriding vars I did not want to. how were you able to not use `:=` for `err`? (only `commonClient` is prior declared, right?)

Reply:
`c` and `err` are declared as return variables at the top of the function, so they don't need to be declared in func body, and they're not needed in the return call. See https://go.dev/tour/basics/7 for example usage.

Reply:
nicee, Thankss!

Reply:
But yes, `:=` in a block scope will override any variables declared outside that scope.
```
foo := 1
if err != nil {
    foo, bar := myFunc()
}
// foo is still 1
foo, bar := myFunc2()
// now foo is the first return value of `myFunc2` instead of 1, since := was used in the same scope as foo was previously defined
```

Reply:
yes, I had I lot of trouble with that in my first Go codes heheh